src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/SemanticTokens/RemoteSemanticTokensLegendService.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor.SemanticTokens;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor.SemanticTokens;
+
+[Export(typeof(ISemanticTokensLegendService)), Shared]
+internal sealed class RemoteSemanticTokensLegendService : ISemanticTokensLegendService
+{
+    public SemanticTokenModifiers TokenModifiers { get; private set; } = null!;
+
+    public SemanticTokenTypes TokenTypes { get; private set; } = null!;
+
+    // TODO: This is dodgy! Need some way to initialize data
```

Comment:
Agree this is dodgy! I was about to add a comment to the properties above and stopped when I saw this comment.

As you mentioned in the PR description, there are a fair number of TODOs, which is to be expected. Is there someplace you're tracking these to make sure you get to them all?

Reply:
I have a laundry list. Will put them all in an issue before merging this PR, I promise!

Though at least this one is easy, once https://github.com/dotnet/razor/pull/10098 is in.

Reply:
https://github.com/dotnet/razor/issues/10103